src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/services/ingest.js

```diff
@@ -0,0 +1,58 @@
+import uiModules from 'ui/modules';
+import _  from 'lodash';
+import angular from 'angular';
+import { keysToCamelCaseShallow, keysToSnakeCaseShallow } from '../../../../../../../common/lib/case_conversion';
+
+const app = uiModules.get('kibana');
+
+app.service('ingest', function ($http) {
+  return {
+    simulatePipeline: simulatePipeline
+  };
+
+  function packagePipeline(pipeline) {
+    const requiredFields = ['input', 'processors'];
+    const cleanedPipeline = _.pick(pipeline, requiredFields);
+    return keysToSnakeCaseShallow(cleanedPipeline);
+  }
+
+  function packageProcessor(processor) {
+    const uiFields = [
+      '$$hashKey',
+      'collapsed',
+      'description',
+      'error',
+      'getDescription',
+      'inputObject',
+      'outputObject',
+      'parent',
+      'setParent',
+      'title',
+      'updateDescription'
+    ];
+
+    const cleanedProcessor = _.omit(processor, uiFields);
+    return keysToSnakeCaseShallow(cleanedProcessor);
+  }
+
+  function packageRequest(pipeline) {
+    const apiPipeline = packagePipeline(pipeline);
+    apiPipeline.processors = apiPipeline.processors.map((processor) => packageProcessor(processor));
```

Comment:
No need for the anonymous function inside map, you can just call it like `apiPipeline.processors.map(packageProcessor);`